receiver/awscontainerinsightreceiver/internal/neuron/pod_attribute_decorator.go

```diff
@@ -0,0 +1,142 @@
+	rms := md.ResourceMetrics()
+	for i := 0; i < rms.Len(); i++ {
+		rs := rms.At(i)
+		ilms := rs.ScopeMetrics()
+		for j := 0; j < ilms.Len(); j++ {
+			ils := ilms.At(j)
+			metrics := ils.Metrics()
+
+			neuronHardwareInfo := pmetric.Metric{}
+			neuronHardwareInfoFound := false
+			for k := 0; k < metrics.Len(); k++ {
+				m := metrics.At(k)
+				if m.Name() == neuronHardwareInfoKey {
+					neuronHardwareInfo = m
+					neuronHardwareInfoFound = true
+					break
+				}
+			}
+			if neuronHardwareInfoFound {
+				neuronCoresPerDevice := getNueronCoresPerDevice(neuronHardwareInfo)
+				for k := 0; k < metrics.Len(); k++ {
+					m := metrics.At(k)
+					pdc.addPodCorrelationAttributes(getMetricDatapoints(m), neuronCoresPerDevice)
+				}
+			}
+		}
+	}
+}
+
+func (pdc *PodAttributesDecoratorConsumer) addPodCorrelationAttributes(metricDatapoints pmetric.NumberDataPointSlice, neuronCoresPerDevice int) {
+	for i := 0; i < metricDatapoints.Len(); i++ {
+		attributes := metricDatapoints.At(i).Attributes()
+		var containerInfo *stores.ContainerInfo
+
+		if neuronDeviceIndex, neuronDeviceIndexPresent := attributes.Get(neuronDeviceAttributeKey); neuronDeviceIndexPresent {
+			// get container info from neuronDeviceIndex
+			neuronDeviceIndex := neuronDeviceIndex.AsString()
+			containerInfo = pdc.getContainerInfoForNeuronDeviceIndex(neuronDeviceIndex)
+
+		} else if neuronCoreIndex, neuronCoreIndexPresent := attributes.Get(neuronCoreAttributeKey); neuronCoreIndexPresent {
+			// get container info from neuronCore
+			containerInfo = pdc.PodResourcesStore.GetContainerInfo(neuronCoreIndex.AsString(), neuronCoreResourceName)
+			neuronDeviceIndex := getNeuronDeviceIndexFromCoreAttribute(neuronCoreIndex, neuronCoresPerDevice)
+			if containerInfo == nil {
+				// else get container info from calculated neuronDeviceIndex
+				containerInfo = pdc.getContainerInfoForNeuronDeviceIndex(neuronDeviceIndex)
+			}
+			attributes.PutStr(neuronDeviceAttributeKey, neuronDeviceIndex)
+		}
+		populateAttributes(&attributes, containerInfo)
+	}
+}
+
+func (pdc *PodAttributesDecoratorConsumer) getContainerInfoForNeuronDeviceIndex(neuronDeviceIndex string) *stores.ContainerInfo {
+	containerInfo := pdc.PodResourcesStore.GetContainerInfo(neuronDeviceIndex, neuronDeviceResourceName)
+	if containerInfo == nil {
+		// Alt resource name is to support backward compatibility in neuron monitor : https://awsdocs-neuron.readthedocs-hosted.com/en/latest/containers/tutorials/k8s-setup.html
+		containerInfo = pdc.PodResourcesStore.GetContainerInfo(neuronDeviceIndex, neuronDeviceResourceNameAlt)
+	}
+	return containerInfo
+}
+
+func populateAttributes(attributes *pcommon.Map, containerInfo *stores.ContainerInfo) {
+	if containerInfo != nil {
+		attributes.PutStr(ci.AttributeContainerName, containerInfo.ContainerName)
+		attributes.PutStr(ci.AttributeK8sPodName, containerInfo.PodName)
+		attributes.PutStr(ci.AttributePodName, containerInfo.PodName)
```

Comment:
Sorry forgot to mention this last time, PodName should also not be done here, the K8sDecorator will populate it.  Confusingly, what Container Insights calls PodName is actually more like workload name.

Reply:
Ohh I see, i will remove the Podname from here then